walk/walk_test.go

```diff
@@ -166,22 +166,31 @@ func TestExcludeFiles(t *testing.T) {
 		{
 			Path: "BUILD.bazel",
 			Content: `
+# gazelle:exclude **/*.pb.go
+# gazelle:exclude *.gen.go
```

Comment:
Let's also test a directory.

```
# gazelle:exclude a/**/b
```

Should exclude `a/b/bad`, and `a/x/b/bad`.

Reply:
done